### PR TITLE
Add stdin and multidoc support to preflight.

### LINKED ISSUE #1109: Allow `preflight`binary to accept stdin as well as URI and files for spec input
The `preflight` binary currently does not accept stdin as a method to pass yaml documents containing Troubleshoot specs.

This task is to add that ability.

Definition of done:

- [x] `preflight` accepts stdin such that `cat mypreflight.yaml | kubectl preflight` produces the output of preflights from mypreflight.yaml
- [x] the content of stdin can be multiple yaml documents, each containing a valid preflight(s), results are from a merged spec
- [ ] Docs are updated in troubleshoot.sh and/or docs.replicated.com

Out of scope:
- Ability to mix stdin and other URI types of input

### PR DESCRIPTION
**Pull request by danj-replicated**
## Description, Motivation and Context

Please include a summary of the change or what problem it solves. Please also include relevant motivation and context.

partially addresses #1111 
Fixes #1109 

Preflight can now accept stdin alongside existing documents like so:

```bash
helm template | preflight -
```

input documents are filtered for valid yaml with the key-value pair: `kind: Preflight`
## Checklist

- [ ] New and existing tests pass locally with introduced changes.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/preflight/run.go [hunk 1]
```diff
@@ -118,27 +125,48 @@ func RunPreflights(interactive bool, output string, format string, args []string
...
+			preflightContent, err = docrewrite.ConvertToV1Beta2([]byte(doc))
+			if err != nil {
+				return errors.Wrap(err, "failed to convert to v1beta2")
+			}
+
+			troubleshootclientsetscheme.AddToScheme(scheme.Scheme)
+			decode := scheme.Codecs.UniversalDeserializer().Decode
+			obj, _, err := decode([]byte(preflightContent), nil, nil)
+			if err != nil {
+				return errors.Wrapf(err, "failed to parse %s", v)
```
Comment (banjoh) [inline pkg/preflight/run.go:156]: Decoding is bound to fail here since troubleshoot will not know how to decode custom CRDs

Comment (danj-replicated) [reply pkg/preflight/run.go:156]: I don't think custom CRDs will get to this point as we skip anything that's not `kind: Preflight`

**Comment by banjoh** [comment]
I ran a test using `cat spec.yaml | ./bin/preflight -`. Works like a charm. This change also brings in a good side effect. The `preflight` binary now allows using a multidoc spec

**Comment by banjoh** [comment]
`cat spec.yaml | ./bin/preflight - spec-2.yaml` works too

**Comment by banjoh** [comment]
An integration test would be good now

**Review by xavpaice — APPROVED** [review]
Looks good, tests are written

**Pull request merged**